dev/com.ibm.ws.jca.cm/src/com/ibm/ws/jca/cm/AbstractConnectionFactoryService.java

```diff
@@ -418,6 +427,34 @@ public XAResource getXAResource(Serializable xaresinfo) throws XAResourceNotAvai
 
             // TODO supply class loader identifier if resource was loaded from application
             ManagedConnectionFactory mcf = getManagedConnectionFactory(null);
+
+            if (qmidenabled) {
```

Comment:
Move this entire block to ConnectionFactoryService.  You will need to add new abstract method to accomplish this, which will be no-op for DataSourceService.

Reply:
Created new method